src/App.tsx

```diff
@@ -1,14 +1,57 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+export enum FilterStatus {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isloading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.all);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos
+      .filter((todo: Todo) => {
+        switch (filterStatus) {
+          case FilterStatus.active:
+            return !todo.completed;
+
+          case FilterStatus.completed:
+            return todo.completed;
+
+          default:
+            return todo;
+        }
+      })
+      .filter(todo =>
+        todo.title.toLowerCase().trim().includes(query.trim().toLowerCase()),
+      );
```

Comment:
It's very hard to read.
better 
const normalizeString = (str: string): string => str.toLowerCase().trim();// separate function

const queryNormalized = normalizeString(query);
const result = normalizeString(todo.title).includes(queryNormalized);